lib/reverse_sentence.rb

```diff
@@ -1,6 +1,40 @@
-# A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
+
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+  length = my_sentence.length
+
+  reverse(my_sentence, 0, length - 1)
+  reverse_words(my_sentence)
+  return
+end
+
+def reverse_words(my_words)
+  return nil if my_words == nil
+
+  start_index = 0
+  end_index = 0
+
+  my_words.length.times do |i|
+    end_index = i
```

Comment:
Good use of detailed variable names! 😃

(Of course, my standard "this is personal preference" disclaimer applies. Some people prefer being concise, others prefer being explicit. 🙂)